src/supertux/menu/editor_delete_levelset_menu.cpp

```diff
@@ -72,14 +72,17 @@ EditorDeleteLevelsetMenu::menu_action(MenuItem& item)
 
   if (id >= 0)
   {
-    std::vector<std::string>& contrib_worlds = m_editor_levelset_select_menu->get_contrib_worlds();
-    if (Editor::is_active() && Editor::current()->get_world() && Editor::current()->get_world()->get_basedir() == contrib_worlds[id])
+    if (Editor::is_active() && Editor::current()->get_world() && Editor::current()->get_world()->get_basedir() == m_editor_levelset_select_menu->get_contrib_worlds().at(id))
       Dialog::show_message(_("You cannot delete the world that you are editing"));
     else
     {
-      physfsutil::remove_with_content(contrib_worlds[id]);
-      m_editor_levelset_select_menu->reload_menu();
-      refresh();
+      Dialog::show_confirmation(_("Are you sure?"), [this, id]()
+      {
+        std::vector<std::string>& contrib_worlds = m_editor_levelset_select_menu->get_contrib_worlds();
+        physfsutil::remove_with_content(contrib_worlds[id]);
```

Comment:
It's probably the most complex logic about a variable's lifetime I've done in a while, but:

`contrib_worlds` can be passed as capture as well, by reference, like `[this, id, &contrib_worlds]`". Although `contrib_worlds` is a local variable, and by the time it'll be executed the variable will have ended its life, it can be noted that this is a reference, and the list it refers to will still exist when the lambda will be executed.

Also, `contrib_worlds` can be declared as `const`, and you may use `auto` to avoid typing `std::vector<std::string>` at long.